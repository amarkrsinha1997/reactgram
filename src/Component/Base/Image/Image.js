import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import './Image.css'

const imageBaseUrl = "https://raw.githubusercontent.com/pankajladhar/Reactgram/master/app/images";

export const Image = ({ imageName, alt, ...others }) => {
	const url = `${imageBaseUrl}/${imageName}`
	return (
		<Fragment>
			<img className="image" data-testid={imageName} src={url} alt={alt} {...others} />
		</Fragment>
	)
}

Image.propTypes = {
	imageName: PropTypes.string.isRequired,
	alt: PropTypes.string.isRequired
}