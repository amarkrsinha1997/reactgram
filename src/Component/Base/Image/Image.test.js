import React from 'react';
import { render, screen } from '@testing-library/react';
import { Image } from './Image';

test('renders with required props', () => {
    const { container } = render(<Image alt="image" imageName="image.jpg"/>);
    expect(container).toMatchSnapshot();
});

test('renders with extra props passed like height, width', () => {
    const { container } = render(<Image alt="image" imageName="image.jpg" height="20px" width="20px"/>);
    const imgElement = screen.getByTestId(/image.jpg/);
    expect(imgElement).toHaveAttribute('height', '20px')
    expect(imgElement).toHaveAttribute('width', '20px')
    expect(container).toMatchSnapshot();
});
