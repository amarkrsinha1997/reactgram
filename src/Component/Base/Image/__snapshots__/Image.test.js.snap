// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with extra props passed like height, width 1`] = `
<div>
  <img
    alt="image"
    class="image"
    data-testid="image.jpg"
    height="20px"
    src="https://raw.githubusercontent.com/pankajladhar/Reactgram/master/app/images/image.jpg"
    width="20px"
  />
</div>
`;

exports[`renders with required props 1`] = `
<div>
  <img
    alt="image"
    class="image"
    data-testid="image.jpg"
    src="https://raw.githubusercontent.com/pankajladhar/Reactgram/master/app/images/image.jpg"
  />
</div>
`;
